src/components/Tooltip/TooltipRenderedOnPageBody.js

```diff
@@ -39,59 +33,92 @@ const propTypes = {
     A positive value shifts the tooltip down, and a negative value shifts it up. */
     shiftVertical: PropTypes.number.isRequired,
 
-    /** Callback to set the Ref to the Tooltip */
-    setTooltipRef: PropTypes.func.isRequired,
-
     /** Text to be shown in the tooltip */
     text: PropTypes.string.isRequired,
 
-    /** Callback to be used to calulate the width and height of tooltip */
-    measureTooltip: PropTypes.func.isRequired,
-};
+    /** Number of pixels to set max-width on tooltip  */
+    maxWidth: PropTypes.number.isRequired,
 
-const defaultProps = {};
-
-const TooltipRenderedOnPageBody = (props) => {
-    const {
-        animationStyle,
-        tooltipWrapperStyle,
-        tooltipTextStyle,
-        pointerWrapperStyle,
-        pointerStyle,
-    } = getTooltipStyles(
-        props.animation,
-        props.windowWidth,
-        props.xOffset,
-        props.yOffset,
-        props.wrapperWidth,
-        props.wrapperHeight,
-        props.tooltipWidth,
-        props.tooltipHeight,
-        props.shiftHorizontal,
-        props.shiftVertical,
-    );
-    return ReactDOM.createPortal(
-        <Animated.View
-            ref={props.setTooltipRef}
-            onLayout={props.measureTooltip}
-            style={[tooltipWrapperStyle, animationStyle]}
-        >
-            <Text style={tooltipTextStyle} numberOfLines={1}>{props.text}</Text>
-            <View style={pointerWrapperStyle}>
-                <View style={pointerStyle} />
-            </View>
-        </Animated.View>,
-        document.querySelector('body'),
-    );
+    /** Maximum number of lines to set on tooltip */
+    numberOfLines: PropTypes.number.isRequired,
 };
 
-TooltipRenderedOnPageBody.propTypes = propTypes;
-TooltipRenderedOnPageBody.defaultProps = defaultProps;
-TooltipRenderedOnPageBody.displayName = 'TooltipRenderedOnPageBody';
-
 // Props will change frequently.
 // On every tooltip hover, we update the position in state which will result in re-rendering.
 // We also update the state on layout changes which will be triggered often.
 // There will be n number of tooltip components in the page.
 // Its good to memorize this one.
-export default memo(TooltipRenderedOnPageBody);
+class TooltipRenderedOnPageBody extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            // tooltipTextWidth will update with wrapped text offsetWidth based on maxWidth prop
+            tooltipTextWidth: this.props.maxWidth,
+
+            // The width and height of the tooltip itself
+            tooltipWidth: 0,
+            tooltipHeight: 0,
+        };
+
+        this.measureTooltip = this.measureTooltip.bind(this);
+    }
+
+    componentDidMount() {
+        this.setState({
+            tooltipTextWidth: this.textRef.offsetWidth,
```

Comment:
I feel like we can get this value from the `this.state.tooltipWidth`.

tooltipTextWidth = this.state.tooltipWidth - (spacing.ph2.paddingHorizontal * 2);

Reply:
So if there is a race condition then why aren't we using the onLayout on the Text or [measure](https://reactnative.dev/docs/0.67/direct-manipulation#measurecallback).

Reply:
Measure returns the same width values as ref.offsetWidth , actually, if you use measure and use ‘max-width’ css prop in wrapperStyle, you can reproduce the bug easily where the blackbox tooltip won’t be taking the full max-width. The same happens with measureInWindow.

And onLayout does not work on nested <Text> components.

Reply:
The only reason to keep pushing this is that something does not feel right. Using `offsetWidth` is not the standard way in react-native. It is vanilla Js. 

But I am fine with it as far as it is working.

Reply:
Ok, I understand now.  The bug where the blackbox tooltip doesn't take the full max-width came form `max-width` css prop. As long as we use `width` instead of `max-width` in wrapperStyle there is no difference between using `measure` or `offsetWidth`. So I can change it to `measure` to be on sync with the react-native way.